simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java

```diff
@@ -107,34 +110,109 @@ protected static boolean shouldUseCompression(HttpExchange exchange) {
         return names;
     }
 
-    protected HttpServer server;
+
+    static class CustomThreadFactory implements ThreadFactory {
+        private ThreadFactory delegate;
+        private final boolean daemon;
+
+        CustomThreadFactory(ThreadFactory delegate, boolean daemon) {
+            this.delegate = delegate;
+            this.daemon = daemon;
+        }
+
+        @Override
+        public Thread newThread(Runnable r) {
+            Thread t = delegate.newThread(r);
+            if (daemon ^ t.isDaemon()) {
```

Comment:
why not always setDaemon?

Reply:
It does not matter in this case. I just looked through the code of DefaultThreadFactory and they use similar code: 
```java
  if (t.isDaemon())
    t.setDaemon(false);
```
Will change to setDaemon since it looks  simpler.